src/progress_bar_exercise/ProgressBar/progress-bar.component.js

```diff
@@ -0,0 +1,51 @@
+import { useCss } from "kremling";
+import React, { useEffect, useState } from "react";
+
+export default function ProgressBar({ loading }) {
+  const [seconds, setSeconds] = useState(0)
+  const [opacity, setOpacity] = useState(1)
+  const [visible, setVisible] = useState()
+
+  useEffect(() => {
+    if (seconds && !loading) { // loading was toggled to false and we have a count going, i.e., request finished
+      setSeconds(16)
+      const timer = setTimeout(() => setOpacity(0), 3000)
+      return () => clearTimeout(timer)
+    } else if (loading) { // loading is true
+      if (seconds > 15 || seconds === 0) setSeconds(1)
+      setOpacity(1)
+      const interval = setInterval(() => {
+        if (seconds < 15) setSeconds(seconds + 1)
+      }, 1000)
+      return () => clearInterval(interval)
+    }
+  }, [loading, seconds, opacity])
+
+  useEffect(() => {
+    // the purpose of something like the "visible" state variable might be to unmount the progress bar completely
```

Comment:
This is a callout that belonged in other PR but I've scrambled things a bit, apologies.